### PR TITLE
Make torch.eye_ init retain requires_grad as original tensor

### LINKED ISSUE #8692: nn.init.eye_ sets requires_grad to False on tensors
repro:
```
import torch.nn as nn
L = nn.Linear(5,5)
nn.init.eye_(L.weight)
print(L.weight.requires_grad)  # False
```

Initially reported from https://discuss.pytorch.org/t/since-0-4-using-nn-init-eye-disables-gradient/19999/2

### PR DESCRIPTION
**Pull request by vishwakftw**
This fixes #8692 

cc: @zou3519

### CODE REVIEW

**Review by soumith — APPROVED** [review]

**Review by zou3519 — APPROVED** [review]

**Comment by bhavesh2109** [comment]
Hey. All the initialisers use the torch.no_grad method before returning. This makes me wonder if it was intended? And if not, then why are the initialisers using the torch.no_grad at all?

**Comment by vishwakftw** [comment]
@bhavesh2109 Please look at the comment in the issue.
